### PR TITLE
pulseaudio: hsphfpd implementation

### LINKED ISSUE #87910: Package hsphfpd & pali’s pulseaudio patches
**Project description**
It would be nice to have hsphfpd and PulseAudio variant that includes support for hsphfpd and patchset from: https://gitlab.freedesktop.org/pulseaudio/pulseaudio/-/merge_requests/288

This should allow, among other things, to control the system volume from headphones.

Relevant issue:

* https://gitlab.freedesktop.org/pulseaudio/pulseaudio/-/issues/722

Other interesting patches are:

* Improved codecs: https://gitlab.freedesktop.org/pulseaudio/pulseaudio/-/merge_requests/227
* https://gitlab.freedesktop.org/pulseaudio/pulseaudio/-/merge_requests/254#note_479030
* absolute volume: https://gitlab.freedesktop.org/pulseaudio/pulseaudio/-/merge_requests/239#note_479656

**Metadata**

* homepage URL: https://github.com/pali/hsphfpd-prototype
* license: confirmed via IRC: “for my perl projects I'm using standard perl license, see: <https://dev.perl.org/licenses/>”
* platforms: linux

### PR DESCRIPTION
**Pull request by eadwu**
###### Motivation for this change
Only meant as a reference implementation for testing

Closes: https://github.com/NixOS/nixpkgs/issues/87910

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/os-specific/linux/bluez/default.nix [hunk 1]
```diff
@@ -26,6 +26,11 @@ in stdenv.mkDerivation rec {
     sha256 = "1p2ncvjz6alr9n3l5wvq2arqgc7xjs6dqyar1l9jp0z8cfgapkb8";
   };
 
+  patches = [
+    # ./config-after-first-failure.patch
```
Comment (SuperSandro2000) [inline pkgs/os-specific/linux/bluez/default.nix:30]: Is this a misstake?

Comment (eadwu) [reply pkgs/os-specific/linux/bluez/default.nix:30]: Read the description, it's a reference implementation, as well as the root MR request, https://gitlab.freedesktop.org/pulseaudio/pulseaudio/-/merge_requests/288#note_672321

I'm not planning on ever getting this into shape, I can't tell any functionality difference between this over the normal pulseaudio, actually causes more instability for me.

Comment (jtojnar) [reply pkgs/os-specific/linux/bluez/default.nix:30]: I will try to work on this, if I find time. One benefit would be having support for absolute volume levels, especially in conjunction with [!239](https://gitlab.freedesktop.org/pulseaudio/pulseaudio/-/merge_requests/239#note_479656). Another would be bluetooth battery indicator but that still needs upower support.

**Comment by jtojnar** [comment]
@eadwu Enabling debug mode should be something like the following options:

```nix
hardware.pulseaudio.daemon.logLevel = "debug";
systemd.services.bluetooth.serviceConfig.ExecStart = [ "" "${hardware.bluetooth.package}/libexec/bluetooth/bluetoothd -d" ];
```

**Comment by gebner** [comment]
I believe we can get this in a mergable state by adding a second duplicate pulseaudio derivation for the hsphfpd-patched version.  Then we don't need to modify the default pulseaudio and don't cause any mass rebuilds.  I'm currently running this and it seems to work fine.

@eadwu Is it ok if I take over the PR and make changes along those lines?  We want to get the hsphfpd part merged anyhow since apparently pipewire can use it as well now.

**Comment by eadwu** [comment]
Works fine for me.

The reason why I used meson in the first place was because the autotools build was broken on my end.

**Comment by jtojnar** [comment]
Hopefully, Pulseaudio 14 will be released before the end of the year and we will be able to switch to Meson in the main derivation.

https://gitlab.freedesktop.org/pulseaudio/pulseaudio/-/issues/1007

**Comment by gebner** [comment]
New PR: #103225

**Pull request closed without merging**